flocker/node/agents/test/test_blockdevice.py

```diff
@@ -844,6 +844,15 @@ class IBlockDeviceAPITestsMixin(object):
     """
     Tests to perform on ``IBlockDeviceAPI`` providers.
     """
+    def tearDown(self):
+        """
+        Destroy volumes created for the cluster used by the test.
+        """
+        volumes = self.api.list_volumes()
+
+        for volume in volumes:
+            self.api.destroy_volume(volume.blockdevice_id)
```

Comment:
Here's what I mean.

```
(1762)(EBS-IBlockDeviceAPI-FLOC-1762 ? =)[~/.../HybridLogic/flocker]$ sudo $(type -p trial) flocker.node.agents.test
:0: UserWarning: You do not have a working installation of the service_identity module: 'No module named service_identity'.  Please install it from <https://pypi.python.org/pypi/service_identity> and make sure all of its dependencies are satisfied.  Without the service_identity module and a recent enough pyOpenSSL to support it, Twisted can perform only rudimentary TLS client hostname verification.  Many valid certificate/hostname mappings may be rejected.
flocker.node.agents.test.test_blockdevice
  AttachVolumeInitTests
    test_init ...                                                          [OK]
    test_optional_arguments ...                                            [OK]
    test_optional_defaults ...                                             [OK]
  AttachVolumeTests
    test_equality ...                                                      [OK]
    test_interface ...                                                     [OK]
    test_notequality ...                                                   [OK]
    test_run ...                                                           [OK]
  BlockDeviceDeployerAlreadyConvergedCalculateChangesTests
    test_deleted_ignored ...                                          [SKIPPED]
    test_no_changes ...                                                    [OK]
  BlockDeviceDeployerCreationCalculateChangesTests
    test_match_configuration_to_state_of_datasets ...                      [OK]
    test_no_devices_no_local_datasets ...                                  [OK]
    test_no_devices_one_dataset ...                                        [OK]
  BlockDeviceDeployerDestructionCalculateChangesTests
    test_delete_before_resize ...                                          [OK]
    test_deleted_dataset_belongs_to_other_node ...                         [OK]
    test_deleted_dataset_volume_exists ...                                 [OK]
  BlockDeviceDeployerDiscoverStateTests
    test_attached_and_mismounted ...                                       [OK]
    test_attached_unmounted_device ...                                     [OK]
    test_no_devices ...                                                    [OK]
    test_one_device ...                                                    [OK]
    test_only_remote_device ...                                            [OK]
    test_only_unattached_devices ...                                       [OK]
    test_unrelated_mounted ...                                             [OK]
  BlockDeviceDeployerResizeCalculateChangesTests
    test_maximum_size_increased ...                                        [OK]
    test_multiple_resize ...                                               [OK]
  BlockDeviceDeployerTests
    test_calculate_necessary_state_changes ...                             [OK]
    test_discover_state_iclusterstatechange ...                            [OK]
    test_discover_state_list_result ...                                    [OK]
    test_interface ...                                                     [OK]
  BlockDeviceInterfaceTests
    test_names ...                                                         [OK]
    test_same_signatures ...                                               [OK]
  CreateBlockDeviceDatasetInitTests
    test_init ...                                                          [OK]
    test_optional_arguments ...                                            [OK]
    test_optional_defaults ...                                             [OK]
  CreateBlockDeviceDatasetTests
    test_equality ...                                                      [OK]
    test_interface ...                                                     [OK]
    test_notequality ...                                                   [OK]
    test_run_create ... mke2fs 1.42.12 (29-Aug-2014)
                                                   [OK]
    test_run_mkfs_and_mount ... mke2fs 1.42.12 (29-Aug-2014)
                                           [OK]
  CreateFilesystemInitTests
    test_init ...                                                          [OK]
    test_optional_arguments ...                                            [OK]
    test_optional_defaults ...                                             [OK]
  CreateFilesystemTests
    test_equality ...                                                      [OK]
    test_interface ...                                                     [OK]
    test_notequality ...                                                   [OK]
  DestroyBlockDeviceDatasetInitTests
    test_init ...                                                          [OK]
    test_optional_arguments ...                                            [OK]
    test_optional_defaults ...                                             [OK]
  DestroyBlockDeviceDatasetTests
    test_dataset_id_must_be_uuid ...                                       [OK]
    test_destroy_nonexistent ...                                           [OK]
    test_equality ...                                                      [OK]
    test_interface ...                                                     [OK]
    test_notequality ...                                                   [OK]
    test_run ...                                                           [OK]
  DestroyVolumeInitTests
    test_init ...                                                          [OK]
    test_optional_arguments ...                                            [OK]
    test_optional_defaults ...                                             [OK]
  DestroyVolumeTests
    test_equality ...                                                      [OK]
    test_interface ...                                                     [OK]
    test_notequality ...                                                   [OK]
    test_run ...                                                           [OK]
  DetachVolumeInitTests
    test_init ...                                                          [OK]
    test_optional_arguments ...                                            [OK]
    test_optional_defaults ...                                             [OK]
  DetachVolumeTests
    test_equality ...                                                      [OK]
    test_interface ...                                                     [OK]
    test_notequality ...                                                   [OK]
    test_run ...                                                           [OK]
  LoopbackBlockDeviceAPIImplementationTests
    test_create_sparse ...                                                 [OK]
    test_initialise_directories ...                                        [OK]
    test_initialise_directories_attached_exists ...                        [OK]
    test_initialise_directories_unattached_exists ...                      [OK]
    test_list_attached_volumes ...                                         [OK]
    test_list_unattached_volumes ...                                       [OK]
    test_resize_data_preserved ...                                         [OK]
    test_resize_grow_sparse ...                                          [FAIL]
  LoopbackBlockDeviceAPITests
    test_attach_attached_volume ...                                     [ERROR]
    test_attach_destroyed_volume ...                                       [OK]
    test_attach_elsewhere_attached_volume ...                           [ERROR]
    test_attach_unattached_volume ...                                   [ERROR]
    test_attach_unknown_volume ...                                         [OK]
    test_attached_volume_listed ...                                     [ERROR]
    test_created_is_listed ...                                             [OK]
    test_created_volume_attributes ...                                     [OK]
    test_destroy_destroyed_volume ...                                      [OK]
    test_destroy_unknown_volume ...                                        [OK]
    test_destroy_volume ...                                                [OK]
    test_detach_detached_volume ...                                        [OK]
    test_detach_unknown_volume ...                                         [OK]
    test_detach_volume ...                                              [ERROR]
    test_foreign_cluster_volume ...                                     [ERROR]
    test_get_device_path_device ...                                     [ERROR]
    test_get_device_path_device_repeatable_results ...                  [ERROR]
    test_get_device_path_unattached_volume ...                             [OK]
    test_get_device_path_unknown_volume ...                                [OK]
    test_interface ...                                                     [OK]
    test_list_attached_and_unattached ...                               [ERROR]
    test_list_volume_empty ...                                             [OK]
    test_listed_volume_attributes ...                                      [OK]
    test_multiple_volumes_attached_to_host ...                          [ERROR]
    test_reattach_detached_volume ...                                   [ERROR]
    test_resize_destroyed_volume ...                                       [OK]
    test_resize_unknown_volume ...                                         [OK]
    test_resize_volume_listed ...                                          [OK]
  LosetupListTests
    test_parse_empty ...                                                   [OK]
    test_parse_multiple_lines ...                                          [OK]
    test_parse_one_line ...                                                [OK]
    test_remove_deleted_suffix ...                                         [OK]
    test_remove_inode ...                                                  [OK]
  MountBlockDeviceInitTests
    test_init ...                                                          [OK]
    test_optional_arguments ...                                            [OK]
    test_optional_defaults ...                                             [OK]
  MountBlockDeviceTests
    test_equality ...                                                      [OK]
    test_interface ...                                                     [OK]
    test_mountpoint_error ... mke2fs 1.42.12 (29-Aug-2014)
                                             [OK]
    test_mountpoint_exists ... mke2fs 1.42.12 (29-Aug-2014)
                                            [OK]
    test_notequality ...                                                   [OK]
    test_run ... mke2fs 1.42.12 (29-Aug-2014)
                                                          [OK]
  ResizeBlockDeviceDatasetInitTests
    test_init ...                                                          [OK]
    test_optional_arguments ...                                            [OK]
    test_optional_defaults ...                                             [OK]
  ResizeBlockDeviceDatasetTests
    test_dataset_id_must_be_uuid ...                                       [OK]
    test_dataset_id_required ...                                           [OK]
    test_equality ...                                                      [OK]
    test_interface ...                                                     [OK]
    test_notequality ...                                                   [OK]
    test_ordering ...                                                      [OK]
    test_run_grow ... mke2fs 1.42.12 (29-Aug-2014)
e2fsck 1.42.12 (29-Aug-2014)
resize2fs 1.42.12 (29-Aug-2014)
                                                     [OK]
    test_size_must_be_int ...                                              [OK]
    test_size_required ...                                                 [OK]
  ResizeFilesystemInitTests
    test_init ...                                                          [OK]
    test_optional_arguments ...                                            [OK]
    test_optional_defaults ...                                             [OK]
  ResizeFilesystemTests
    test_equality ...                                                      [OK]
    test_grow ... mke2fs 1.42.12 (29-Aug-2014)
e2fsck 1.42.12 (29-Aug-2014)
resize2fs 1.42.12 (29-Aug-2014)
                                                         [OK]
    test_interface ...                                                     [OK]
    test_notequality ...                                                   [OK]
  ResizeVolumeInitTests
    test_init ...                                                          [OK]
    test_optional_arguments ...                                            [OK]
    test_optional_defaults ...                                             [OK]
  ResizeVolumeTests
    test_equality ...                                                      [OK]
    test_interface ...                                                     [OK]
    test_notequality ...                                                   [OK]
    test_run_grow ...                                                      [OK]
  SyncToThreadedAsyncAPIAdapterTests
    test_interface ...                                                     [OK]
  UnmountBlockDeviceInitTests
    test_init ...                                                          [OK]
    test_optional_arguments ...                                            [OK]
    test_optional_defaults ...                                             [OK]
  UnmountBlockDeviceTests
    test_equality ...                                                      [OK]
    test_interface ...                                                     [OK]
    test_notequality ...                                                   [OK]
    test_run ...                                                           [OK]

===============================================================================
[SKIPPED]
This will pass when the deployer is smart enough to know it should not delete things that do not exist.  FLOC-1756.

flocker.node.agents.test.test_blockdevice.BlockDeviceDeployerAlreadyConvergedCalculateChangesTests.test_deleted_ignored
===============================================================================
[FAIL]
Traceback (most recent call last):
  File "/home/richard/projects/HybridLogic/flocker/flocker/node/agents/test/test_blockdevice.py", line 1572, in test_resize_grow_sparse
    (size.actual, size.reported)
  File "/home/richard/.virtualenvs/1762/lib/python2.7/site-packages/twisted/trial/_synctest.py", line 447, in assertEqual
    % (msg, pformat(first), pformat(second)))
twisted.trial.unittest.FailTest: not equal:
a = (0, 150000000000)
b = (4096, 150000000000)


flocker.node.agents.test.test_blockdevice.LoopbackBlockDeviceAPIImplementationTests.test_resize_grow_sparse
===============================================================================
[ERROR]
Traceback (most recent call last):
  File "/home/richard/projects/HybridLogic/flocker/flocker/node/agents/test/test_blockdevice.py", line 854, in tearDown
    self.api.destroy_volume(volume.blockdevice_id)
  File "/home/richard/projects/HybridLogic/flocker/flocker/node/agents/blockdevice.py", line 1011, in destroy_volume
    volume_path.remove()
  File "/home/richard/.virtualenvs/1762/lib/python2.7/site-packages/twisted/python/filepath.py", line 1245, in remove
    os.remove(self.path)
exceptions.OSError: [Errno 2] No such file or directory: '/home/richard/projects/HybridLogic/flocker/_trial_temp/flocker.node.agents.test.test_bl/LoopbackBlockDeviceAPITests/test_attach_attached_volume/vbpUvA/temp/unattached/block-ae47f92a-2868-44e0-87cd-ab74dd84effb'

flocker.node.agents.test.test_blockdevice.LoopbackBlockDeviceAPITests.test_attach_attached_volume
===============================================================================
[ERROR]
Traceback (most recent call last):
  File "/home/richard/projects/HybridLogic/flocker/flocker/node/agents/test/test_blockdevice.py", line 854, in tearDown
    self.api.destroy_volume(volume.blockdevice_id)
  File "/home/richard/projects/HybridLogic/flocker/flocker/node/agents/blockdevice.py", line 1011, in destroy_volume
    volume_path.remove()
  File "/home/richard/.virtualenvs/1762/lib/python2.7/site-packages/twisted/python/filepath.py", line 1245, in remove
    os.remove(self.path)
exceptions.OSError: [Errno 2] No such file or directory: '/home/richard/projects/HybridLogic/flocker/_trial_temp/flocker.node.agents.test.test_bl/LoopbackBlockDeviceAPITests/test_attach_elsewhere_attached_v/HJpGm8/temp/unattached/block-b01168fa-b328-45a7-a322-92f1d1ba7685'

flocker.node.agents.test.test_blockdevice.LoopbackBlockDeviceAPITests.test_attach_elsewhere_attached_volume
===============================================================================
[ERROR]
Traceback (most recent call last):
  File "/home/richard/projects/HybridLogic/flocker/flocker/node/agents/test/test_blockdevice.py", line 854, in tearDown
    self.api.destroy_volume(volume.blockdevice_id)
  File "/home/richard/projects/HybridLogic/flocker/flocker/node/agents/blockdevice.py", line 1011, in destroy_volume
    volume_path.remove()
  File "/home/richard/.virtualenvs/1762/lib/python2.7/site-packages/twisted/python/filepath.py", line 1245, in remove
    os.remove(self.path)
exceptions.OSError: [Errno 2] No such file or directory: '/home/richard/projects/HybridLogic/flocker/_trial_temp/flocker.node.agents.test.test_bl/LoopbackBlockDeviceAPITests/test_attach_unattached_volume/9RUPyn/temp/unattached/block-bc6a050d-ee6b-420d-98f2-42926280b2f3'

flocker.node.agents.test.test_blockdevice.LoopbackBlockDeviceAPITests.test_attach_unattached_volume
===============================================================================
[ERROR]
Traceback (most recent call last):
  File "/home/richard/projects/HybridLogic/flocker/flocker/node/agents/test/test_blockdevice.py", line 854, in tearDown
    self.api.destroy_volume(volume.blockdevice_id)
  File "/home/richard/projects/HybridLogic/flocker/flocker/node/agents/blockdevice.py", line 1011, in destroy_volume
    volume_path.remove()
  File "/home/richard/.virtualenvs/1762/lib/python2.7/site-packages/twisted/python/filepath.py", line 1245, in remove
    os.remove(self.path)
exceptions.OSError: [Errno 2] No such file or directory: '/home/richard/projects/HybridLogic/flocker/_trial_temp/flocker.node.agents.test.test_bl/LoopbackBlockDeviceAPITests/test_attached_volume_listed/NKMYPV/temp/unattached/block-8ef7aa87-9b1a-41ab-af90-1df5c147015a'

flocker.node.agents.test.test_blockdevice.LoopbackBlockDeviceAPITests.test_attached_volume_listed
===============================================================================
[ERROR]
Traceback (most recent call last):
  File "/home/richard/projects/HybridLogic/flocker/flocker/node/agents/test/test_blockdevice.py", line 854, in tearDown
    self.api.destroy_volume(volume.blockdevice_id)
  File "/home/richard/projects/HybridLogic/flocker/flocker/node/agents/blockdevice.py", line 1011, in destroy_volume
    volume_path.remove()
  File "/home/richard/.virtualenvs/1762/lib/python2.7/site-packages/twisted/python/filepath.py", line 1245, in remove
    os.remove(self.path)
exceptions.OSError: [Errno 2] No such file or directory: '/home/richard/projects/HybridLogic/flocker/_trial_temp/flocker.node.agents.test.test_bl/LoopbackBlockDeviceAPITests/test_detach_volume/pqNMFc/temp/unattached/block-75fae810-d4b1-4e3b-92f8-947459752cf8'

flocker.node.agents.test.test_blockdevice.LoopbackBlockDeviceAPITests.test_detach_volume
===============================================================================
[ERROR]
Traceback (most recent call last):
  File "/home/richard/projects/HybridLogic/flocker/flocker/node/agents/test/test_blockdevice.py", line 1364, in test_foreign_cluster_volume
    old_cluster_id = self.api.cluster_id
exceptions.AttributeError: 'LoopbackBlockDeviceAPI' object has no attribute 'cluster_id'

flocker.node.agents.test.test_blockdevice.LoopbackBlockDeviceAPITests.test_foreign_cluster_volume
===============================================================================
[ERROR]
Traceback (most recent call last):
  File "/home/richard/projects/HybridLogic/flocker/flocker/node/agents/test/test_blockdevice.py", line 854, in tearDown
    self.api.destroy_volume(volume.blockdevice_id)
  File "/home/richard/projects/HybridLogic/flocker/flocker/node/agents/blockdevice.py", line 1011, in destroy_volume
    volume_path.remove()
  File "/home/richard/.virtualenvs/1762/lib/python2.7/site-packages/twisted/python/filepath.py", line 1245, in remove
    os.remove(self.path)
exceptions.OSError: [Errno 2] No such file or directory: '/home/richard/projects/HybridLogic/flocker/_trial_temp/flocker.node.agents.test.test_bl/LoopbackBlockDeviceAPITests/test_get_device_path_device/AszLzO/temp/unattached/block-2e4dd101-d9d8-48a4-9a57-577e754f4956'

flocker.node.agents.test.test_blockdevice.LoopbackBlockDeviceAPITests.test_get_device_path_device
===============================================================================
[ERROR]
Traceback (most recent call last):
  File "/home/richard/projects/HybridLogic/flocker/flocker/node/agents/test/test_blockdevice.py", line 854, in tearDown
    self.api.destroy_volume(volume.blockdevice_id)
  File "/home/richard/projects/HybridLogic/flocker/flocker/node/agents/blockdevice.py", line 1011, in destroy_volume
    volume_path.remove()
  File "/home/richard/.virtualenvs/1762/lib/python2.7/site-packages/twisted/python/filepath.py", line 1245, in remove
    os.remove(self.path)
exceptions.OSError: [Errno 2] No such file or directory: '/home/richard/projects/HybridLogic/flocker/_trial_temp/flocker.node.agents.test.test_bl/LoopbackBlockDeviceAPITests/test_get_device_path_device_repe/nG3vlk/temp/unattached/block-d1a532d9-5029-4323-a838-2ee40bf7d6b8'

flocker.node.agents.test.test_blockdevice.LoopbackBlockDeviceAPITests.test_get_device_path_device_repeatable_results
===============================================================================
[ERROR]
Traceback (most recent call last):
  File "/home/richard/projects/HybridLogic/flocker/flocker/node/agents/test/test_blockdevice.py", line 854, in tearDown
    self.api.destroy_volume(volume.blockdevice_id)
  File "/home/richard/projects/HybridLogic/flocker/flocker/node/agents/blockdevice.py", line 1011, in destroy_volume
    volume_path.remove()
  File "/home/richard/.virtualenvs/1762/lib/python2.7/site-packages/twisted/python/filepath.py", line 1245, in remove
    os.remove(self.path)
exceptions.OSError: [Errno 2] No such file or directory: '/home/richard/projects/HybridLogic/flocker/_trial_temp/flocker.node.agents.test.test_bl/LoopbackBlockDeviceAPITests/test_list_attached_and_unattache/CpXTFW/temp/unattached/block-ebcb89a0-296d-4754-af57-8f87da869f8e'

flocker.node.agents.test.test_blockdevice.LoopbackBlockDeviceAPITests.test_list_attached_and_unattached
===============================================================================
[ERROR]
Traceback (most recent call last):
  File "/home/richard/projects/HybridLogic/flocker/flocker/node/agents/test/test_blockdevice.py", line 854, in tearDown
    self.api.destroy_volume(volume.blockdevice_id)
  File "/home/richard/projects/HybridLogic/flocker/flocker/node/agents/blockdevice.py", line 1011, in destroy_volume
    volume_path.remove()
  File "/home/richard/.virtualenvs/1762/lib/python2.7/site-packages/twisted/python/filepath.py", line 1245, in remove
    os.remove(self.path)
exceptions.OSError: [Errno 2] No such file or directory: '/home/richard/projects/HybridLogic/flocker/_trial_temp/flocker.node.agents.test.test_bl/LoopbackBlockDeviceAPITests/test_multiple_volumes_attached_t/x2UKQa/temp/unattached/block-74015beb-0f3a-489b-87f1-df554c57d49d'

flocker.node.agents.test.test_blockdevice.LoopbackBlockDeviceAPITests.test_multiple_volumes_attached_to_host
===============================================================================
[ERROR]
Traceback (most recent call last):
  File "/home/richard/projects/HybridLogic/flocker/flocker/node/agents/test/test_blockdevice.py", line 854, in tearDown
    self.api.destroy_volume(volume.blockdevice_id)
  File "/home/richard/projects/HybridLogic/flocker/flocker/node/agents/blockdevice.py", line 1011, in destroy_volume
    volume_path.remove()
  File "/home/richard/.virtualenvs/1762/lib/python2.7/site-packages/twisted/python/filepath.py", line 1245, in remove
    os.remove(self.path)
exceptions.OSError: [Errno 2] No such file or directory: '/home/richard/projects/HybridLogic/flocker/_trial_temp/flocker.node.agents.test.test_bl/LoopbackBlockDeviceAPITests/test_reattach_detached_volume/ZYsTyj/temp/unattached/block-52967994-a717-4e96-b19c-920fc6463583'

flocker.node.agents.test.test_blockdevice.LoopbackBlockDeviceAPITests.test_reattach_detached_volume
-------------------------------------------------------------------------------
Ran 151 tests in 112.268s

FAILED (skips=1, failures=1, errors=11, successes=138)

```